src/update-feeds/update-feeds.ts

```diff
@@ -168,12 +169,13 @@ export const updateDynamicState = (batch: ReadDapiWithIndexResponsesAndChainId)
   );
 };
 
-export const getFeedsToUpdate = (batch: ReadDapiWithIndexResponsesAndChainId) =>
+export const getFeedsToUpdate = (batch: ReadDapiWithIndexResponsesAndChainId, sponsorWalletMnemonic: string) =>
   batch.map((dapiResponse) => {
     const signedData = getStoreDataPoint(dapiResponse.dataFeed.dataFeedId);
+    const sponsorWallet = ethers.Wallet.fromMnemonic(sponsorWalletMnemonic);
```

Comment:
Why do we need to pass `sponsorWalletMnemonic` in this function? It seems it's just forwarded.

Reply:
Sorry I should have added a comment on this line, it's a placeholder. The mnemonic is needed to derive the datafeed specific wallet, right? I thought we might as well do it in this function so that the derived address can then be used to both clear update timestamps or send transactions.

Reply:
But to derive the sponsor wallet we need the mnemonic (not the wallet) so this is not relevant. I still think it would be better to just pass the mnemonic to the relevant function that needs it instead of forwarding it.

Reply:
I mean that we'd derive the wallet here, we need the derived wallet (address) to both clear update timestamps for feeds that don't need to be updated as well as to submit transactions for feeds that need to be updated, right?

Or do you mean that it would be ok to derive the wallet multiple times so we'd do it separately while clearing update timestamps as well as when submitting txs?

Reply:
I mean we could just pass the `sponsorWalletMnemonic` (or get it from state) in `updateFeeds` function and derive the wallet there. It would tehn be used to both clear the timestamps and submit the update transactions.

Reply:
Ah I think I get what you mean, then we just wouldn't filter feeds based on `shouldUpdate` in `processBatch`.